actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -26,7 +26,68 @@ module TagHelper
       PRE_CONTENT_STRINGS[:textarea]  = "\n"
       PRE_CONTENT_STRINGS["textarea"] = "\n"
 
+      class TagBuilder #:nodoc:
+        include TagHelper
 
+        VOID_ELEMENTS = %w(base  br  col  embed  hr  img  input  keygen  link
+                        meta  param  source  track  wbr).to_set
+
+        VOID_ELEMENTS.merge(VOID_ELEMENTS.map(&:to_sym))
+
+        ELEMENTS = %w(a abbr acronym address applet area article aside audio b
+                   base basefont bdi bdo bgsound big blink blockquote body br
+                   button canvas caption center cite code col colgroup command
+                   content data datalist dd del details dfn dialog dir div dl
+                   dt element em embed fieldset figcaption figure font footer
+                   form frame frameset h1 h2 h3 h4 h5 h6 head header hgroup hr
+                   html i iframe image img input ins isindex kbd keygen label
+                   legend li link listing main map mark marquee menu menuitem
+                   meta meter multicol nav nextid nobr noembed noframes
+                   noscript object ol optgroup option output p param picture
+                   plaintext pre progress q rp rt rtc ruby s samp script
+                   section select shadow small source spacer span strike strong
+                   style sub summary sup table tbody td template textarea tfoot
+                   th thead time title tr track tt u ul var video wbr xmp
+                   ).to_set
```

Comment:
I don't think we want to declare the regular ELEMENTS, because people should be able to use custom tags with this helper as well. We only need the VOID_ELEMENTS defined such that we can do the right thing for BR, HR, and friends. Assume all other tags are open.